File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:


    int maxArea(vector<int>& height) {
        int currMax = -1;
        int s = 0; int e = height.size() - 1;
        int contW = 0;
        while (s <= e){
            contW = (e - s) * min(height[e], height[s]);
            if (contW > currMax){
                currMax = contW;
            }

            if (height[s] < height[e]){
                s++;
            } else {
                e--;
            }
        }
        return currMax;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var stPtr = 0
        var eptr = height.size - 1
        var currArea = 0
        while (stPtr < eptr){
            currArea = max(currArea, min(height[stPtr], height[eptr]) * (eptr - stPtr))
            if (height[stPtr] < height[eptr]){
                while (stPtr < height.size && height[stPtr] < height[eptr]){
                    currArea = max(currArea, min(height[stPtr], height[eptr]) * (eptr - stPtr))
                    stPtr += 1
                }
            } else {
                while (eptr >= 0 && height[eptr] <= height[stPtr]){
                    currArea = max(currArea, min(height[stPtr], height[eptr]) * (eptr - stPtr))
                    eptr -= 1
                }
            }
        }
        return currArea
    }
}